packages/containers/src/AppLoader/AppLoader.connect.js

```diff
@@ -0,0 +1,51 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import get from 'lodash/get';
+import { cmfConnect } from '@talend/react-cmf';
+import { AppLoader } from '@talend/react-components';
+import { appLoaderSaga } from './AppLoader.saga';
+
+/**
+ * This container show the application's loader & bootstrap the app
+ * @param {object} props the component props
+ * @param {boolean} props.loading tell if the app loader should show the loader or the content
+ * @param {object} props.children react element to show
+ */
+export function AppLoaderContainer({ loading, children }) {
+	if (loading) {
+		return <AppLoader />;
+	}
+
+	return children || null;
+}
+
+AppLoaderContainer.displayName = 'AppLoader';
+AppLoaderContainer.propTypes = {
+	children: PropTypes.element,
+	loading: PropTypes.bool,
+};
+
+/**
+ * calculate the loading attribute with the given props
+ * @param {object} state the redux state
+ * @param {object} ownProps the component props
+ */
+export function mapStateToProps(state, ownProps) {
+	const newProps = { loading: false };
+	get(ownProps, 'hasCollections', []).forEach(collectionName => {
```

Comment:
could you use newsProps.loading = get(ownProps, 'hasCollections', []).every(collectionName => state.cmf.collections.has(collectionName))